integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDiagnosticsCompleteAvailableCondition.java

```diff
@@ -0,0 +1,922 @@
+    checkDomainStatusConditionTypeExists(domainUid, domainNamespace1, DOMAIN_STATUS_CONDITION_AVAILABLE_TYPE);
+    // verify the condition Completed type has status True
+    checkDomainStatusConditionTypeHasExpectedStatus(domainUid, domainNamespace1,
+        DOMAIN_STATUS_CONDITION_COMPLETED_TYPE, "True");
+    // verify the condition Available type has status True
+    checkDomainStatusConditionTypeHasExpectedStatus(domainUid, domainNamespace1,
+        DOMAIN_STATUS_CONDITION_AVAILABLE_TYPE, "True");
+    // verify there is no status condition type Failed
+    verifyDomainStatusConditionTypeDoesNotExist(domainUid, domainNamespace1, DOMAIN_STATUS_CONDITION_FAILED_TYPE);
+  }
+
+  /**
+   * Test domain status condition with serverStartPolicy set to ADMIN_ONLY.
+   * Verify the following conditions are generated:
+   * type: Completed, status: true
+   * type: Available, status: true
+   * Verify no Failed type condition generated.
+   * Disabled due to bug.
+   */
+  @Disabled
+  @Order(2)
+  @Test
+  @DisplayName("Test domain events for various successful domain life cycle changes")
+  void testCompleteAvailableConditionWithAdminOnly() {
+
+    logger.info("patch the domain and change the serverStartPolicy to ADMIN_ONLY");
+
+    String patchStr = "[{\"op\": \"replace\",\"path\": \"/spec/serverStartPolicy\", \"value\": \"ADMIN_ONLY\"}]";
+
+    logger.info("Updating domain configuration using patch string: {0}", patchStr);
+    assertTrue(patchDomainCustomResource(domainUid, domainNamespace1, new V1Patch(patchStr),
+        V1Patch.PATCH_FORMAT_JSON_PATCH), "Failed to patch domain");
+
+    logger.info("Checking for admin server pod is up and running");
+    checkPodReadyAndServiceExists(adminServerPodName, domainUid, domainNamespace1);
+
+    // verify all the managed servers are shutdown
+    logger.info("Checking managed server pods were shutdown");
+    for (int i = 1; i <= replicaCount; i++) {
+      checkPodDoesNotExist(managedServerPodNamePrefix + i, domainUid, domainNamespace1);
+    }
+
+    // verify the condition type Completed exists
+    checkDomainStatusConditionTypeExists(domainUid, domainNamespace1, DOMAIN_STATUS_CONDITION_COMPLETED_TYPE);
+    // verify the condition type Available exists
+    checkDomainStatusConditionTypeExists(domainUid, domainNamespace1, DOMAIN_STATUS_CONDITION_AVAILABLE_TYPE);
+    // verify the condition Completed type has status True
+    checkDomainStatusConditionTypeHasExpectedStatus(domainUid, domainNamespace1,
+        DOMAIN_STATUS_CONDITION_COMPLETED_TYPE, "True");
+    // verify the condition Available type has status True
+    checkDomainStatusConditionTypeHasExpectedStatus(domainUid, domainNamespace1,
+        DOMAIN_STATUS_CONDITION_AVAILABLE_TYPE, "False");
+    // verify there is no status condition type Failed
+    verifyDomainStatusConditionTypeDoesNotExist(domainUid, domainNamespace1, DOMAIN_STATUS_CONDITION_FAILED_TYPE);
+
+    // patch the domain back to the original state
```

Comment:
it will be good to put this code in a finally block so that the domain is restored even if this test fails and rest of the tests are unaffected.

Reply:
done